papis/zenodo.py

```diff
@@ -0,0 +1,208 @@
+    ),
+    KeyConversionPair("version", [{"key": "version", "action": None}]),
+    # extra fields
+    KeyConversionPair("keywords", [{"key": "tags", "action": None}]),
+    KeyConversionPair("revision", [{"key": "revision", "action": None}]),
+    KeyConversionPair("links", [{"key": "url", "action": lambda x: x["self"]}]),
+    KeyConversionPair("method", [{"key": "method", "action": get_text_from_html}]),
+]
+
+
+def zenodo_data_to_papis_data(data: Dict[str, Any]) -> Dict[str, Any]:
+    """Converts the dictionary from Zenodo to the conventional papis format.
+
+    :param data: the raw dictionary from Zenodo
+    :return: a new dictionary in a conventional papis form
+    """
+    # Merge metadata into data
+    data.update(data["metadata"])
+    del data["metadata"]
+
+    new_data = papis.document.keyconversion_to_data(key_conversion, data)
+    return new_data
+
+
+def is_valid_record_id(record_id: str) -> bool:
+    """Checks if a record is a valid Zenodo record, first checking its form and then
+    testing against Zenodo
+
+    :param record_id: a Zenodo record id
+    :return: whether the record is valid
+    """
+    record_id = record_id.strip()
+    if not record_id.isdigit():
+        return False
+
+    with papis.utils.get_session() as session:
+        response = session.get(ZENODO_URL.format(record_id=record_id))
+
+    return response.ok
+
+
+def get_data(record_id: str) -> Dict[str, Any]:
+    """Fetches a record from the Zenodo API and processes it with a helper function
+
+    :param record_id: a Zenodo record id
+    :return: a processed zenodo record
+    """
+    with papis.utils.get_session() as session:
+        response = session.get(
+            ZENODO_URL.format(record_id=record_id.strip()),
+            headers={"user-agent": f"papis/{papis.__version__}"},
+        )
+
+    try:
+        json_data = json.loads(response.content.decode())  # type: Dict[str, Any]
+    except json.JSONDecodeError as exc:
+        logger.error("Failed to decode response from Zenodo.", exc_info=exc)
+
+    return json_data
+
+
+class Context(papis.importer.Context):
+    def __init__(self) -> None:
+        super().__init__()
+        self.file_info: Dict[str, Any] = {}
+
+
+class Importer(papis.importer.Importer):
+    """Importer downloading data from a Zenodo ID"""
+
+    ctx: Context
+
+    def __init__(self, uri: str = "") -> None:
+        super().__init__(name="zenodo", uri=uri, ctx=Context())
+
+    @classmethod
+    def match(cls, uri: str) -> Optional[papis.importer.Importer]:
+        if is_valid_record_id(uri):
+            return Importer(uri)
+
+        return None
+
+    @classmethod
+    def match_data(cls, data: Dict[str, Any]) -> Optional["Importer"]:
+        return None
+
+    def fetch_data(self) -> None:
+        zenodo_data = get_data(self.uri)
+        # Build a filename to URL dictionary
+        self.ctx.file_info = {
+            file["key"]: file["links"]["self"] for file in zenodo_data["files"]
+        }
```

Comment:
```suggestion
        self.ctx.file_info = [
            (file["key"], file["links"]["self"]) for file in zenodo_data["files"]
        ]
```
? Not sure there's any difference, so up to you!